BackofficeBundle/EventSubscriber/FieldTypeTypeSubscriber.php

```diff
@@ -79,24 +80,29 @@ protected function checkFieldType(FieldTypeInterface $data, $type, Form $form)
         if (is_null($type) || !array_key_exists($type, $this->options)) {
             return;
         }
-        $keys = array();
-        foreach ($this->options[$type]['options'] as $key => $option) {
-            if (!$data->hasOption($key)) {
-                $fieldOptionClass = $this->fieldOptionClass;
-                /** @var FieldOptionInterface $fieldOption */
-                $fieldOption = new $fieldOptionClass();
-                $fieldOption->setKey($key);
-                $fieldOption->setValue($option['default_value']);
 
-                $data->addOption($fieldOption);
+        if (array_key_exists('options', $this->options[$type])) {
+            $keys = array();
+            foreach ($this->options[$type]['options'] as $key => $option) {
+                if (!$data->hasOption($key)) {
+                    $fieldOptionClass = $this->fieldOptionClass;
+                    /** @var FieldOptionInterface $fieldOption */
+                    $fieldOption = new $fieldOptionClass();
+                    $fieldOption->setKey($key);
+                    $fieldOption->setValue($option['default_value']);
+
+                    $data->addOption($fieldOption);
+                }
+                $keys[] = $key;
             }
-            $keys[] = $key;
-        }
 
-        foreach ($data->getOptions() as $option) {
-            if (!in_array($option->getKey(), $keys)) {
-                $data->removeOption($option);
+            foreach ($data->getOptions() as $option) {
+                if (!in_array($option->getKey(), $keys)) {
+                    $data->removeOption($option);
+                }
             }
+        } else {
+            $data->setOptions(new ArrayCollection());
```

Comment:
this should not be done here!

The setOption method should not be called to but a empty arrayCollection inside.